pkg/controller/machine/machine_controller.go

```diff
@@ -278,6 +288,45 @@ func (r *ReconcileMachine) isDeleteAllowed(machine *clusterv1.Machine) bool {
 	return node.UID != machine.Status.NodeRef.UID
 }
 
+var (
+	errNilNodeRef           = errors.New("noderef is nil")
+	errLastControlPlaneNode = errors.New("last control plane member")
+	errNoControlPlaneNodes  = errors.New("no control plane members")
+)
+
+// isDeleteNodeAllowed returns nil only if the Machine's NodeRef is not nil
+// and if the Machine is not the last control plane node in the cluster.
+func (r *ReconcileMachine) isDeleteNodeAllowed(ctx context.Context, machine *clusterv1.Machine) error {
+
```

Comment:
Nit: extra space

Reply:
It's a preference thing. I prefer to leave a space when the first line is a comment. If the project style is to not do this, I will remove it.